lib/media/media_source_engine.js

```diff
@@ -1789,6 +1826,81 @@ shaka.media.MediaSourceEngine = class {
     return true;
   }
 
+  /**
+   * Returns true if it's necessary codec switch to load the new stream.
+   *
+   * @param {shaka.util.ManifestParserUtils.ContentType} contentType
+   * @param {shaka.extern.Stream} stream
+   * @return {boolean}
+   * @private
+   */
+  isNecessaryCodecSwitch_(contentType, stream) {
+    const MimeUtils = shaka.util.MimeUtils;
+    const currentCodec = MimeUtils.getCodecBase(
+        MimeUtils.getCodecs(this.sourceBufferTypes_[contentType]));
+    const currentBasicType = MimeUtils.getBasicType(
+        this.sourceBufferTypes_[contentType]);
+
+    /** @type {?shaka.extern.Transmuxer} */
+    let transmuxer;
```

Comment:
Is there any reason you are defining the local variable here, when it is only used inside the 
`if (transmuxerPlugin) {}` clause below?

Reply:
Changed!